azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -331,6 +332,71 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				Optional: true,
 			},
 
+			"maintenance_config": {
```

Comment:
i think config is redundant here, this is more a window?
```suggestion
			"maintenance_window": {
```